tests/integration/framework/component/mixer/mixer_binary_comp.go

```diff
@@ -0,0 +1,104 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mixer
+
+import (
+	"fmt"
+	"log"
+	"os"
+
+	"path/filepath"
+
+	"istio.io/istio/tests/integration/framework"
+	"istio.io/istio/tests/util"
+)
+
+// LocalComponent is a component of local mixs binary in process
+type LocalComponent struct {
+	framework.Component
+	name      string
+	process   *os.Process
+	logFile   string
+	configDir string
+}
+
+// NewLocalComponent create a LocalComponent with name, log dir and config dir
+func NewLocalComponent(n, logDir, configDir string) *LocalComponent {
+	logFile := fmt.Sprintf("%s/%s.log", logDir, n)
+	return &LocalComponent{
+		name:      n,
+		logFile:   logFile,
+		configDir: configDir,
+	}
+}
+
+// GetName return component name
+func (mixerComp *LocalComponent) GetName() string {
+	return mixerComp.name
+}
+
+// Start brings up a local mixs using test config files in local file system
+func (mixerComp *LocalComponent) Start() (err error) {
+	if _, err = util.Shell("bazel build -c opt mixer/cmd/server:mixs"); err != nil {
```

Comment:
I think the component should not need to worry about building.
Building should happen before an environment can run or can be tested. WDYT?

And bazel run should kick off the build first, right?

Reply:
We directly use binary so bazel run will not trigger build. But you are right, we can assume things are built before running integration tests.